src/components/SQLEditor.tsx

```diff
@@ -0,0 +1,75 @@
+import { SQLEditor as SQLCodeEditor } from '@grafana/experimental';
+import { DataSource } from '../DataSource';
+import { getTimestreamCompletionProvider } from 'language/completionItemProvider';
+import { DATABASE_MACRO, TABLE_MACRO } from 'language/macros';
+import React, { useRef, useMemo, useCallback, useEffect } from 'react';
+import { TimestreamQuery } from 'types';
+import timestreamLanguageDefinition from 'language/definition';
+
+interface RawEditorProps {
+  query: TimestreamQuery;
+  onRunQuery: () => void;
+  onChange: (q: string) => void;
+  datasource: DataSource;
+}
+
+export default function SQLEditor({ query, datasource, onRunQuery, onChange }: RawEditorProps) {
+  const queryRef = useRef<TimestreamQuery>(query);
+  useEffect(() => {
+    queryRef.current = query;
+  }, [query]);
+
+  const getDatabases = useCallback(async () => {
+    const databases: string[] = await datasource.postResource('databases');
+    return databases.map((database) => ({ name: database, completion: database }));
+  }, [queryRef.current]);
+
+  const getTables = useCallback(
+    async (database?: string) => {
+      const tables: string[] = await datasource
+        .postResource('tables', {
+          database: database ?? queryRef.current.database ?? '',
+        })
+        .catch(() => []);
+      return tables.map((table) => ({ name: table, completion: table }));
+    },
+    [datasource]
+  );
+
+  const getColumns = useCallback(async (database?: string, tableName?: string) => {
+    const interpolatedArgs = {
+      database: database ? database.replace(TABLE_MACRO, queryRef.current.table ?? '') : queryRef.current.table,
```

Comment:
nit picky feel free to ignore: we're using a lot of `?` here and it's getting a bit hard to read.  Is there a way we could do this on multiple lines or something like that?